tests/integration/optimizers/test_optimizer.py

```diff
@@ -58,14 +58,86 @@ def test_optimizer(tmpdir, config):
     validate_result(result, tmpdir)
 
 
+def test_optimizer_multi_flow(tmpdir, config):
+    multi_flow_runner = MultiFlowRunner(
+        SingleFlowRunner(
+            flow_yaml='tests/integration/optimizers/flow1.yml',
+            documents=document_generator(10),
+            request_size=1,
+            execution_method='search',
+        ),
+        SingleFlowRunner(
+            flow_yaml='tests/integration/optimizers/flow2.yml',
+            documents=document_generator(10),
+            request_size=1,
+            execution_method='search',
+        )
+    )
+    opt = FlowOptimizer(
+        flow_runner=multi_flow_runner,
+        parameter_yaml='tests/integration/optimizers/parameter.yml',
+        evaluation_callback=MeanEvaluationCallback(),
+        workspace_base_dir=str(tmpdir),
+        n_trials=5,
+    )
+    result = opt.optimize_flow()
+    validate_result(result, tmpdir)
+
+
+def test_yaml_multi_flow(tmpdir, config):
+    jsonlines_file = os.path.join(tmpdir, 'docs.jsonlines')
+    optimizer_yaml = f'''!FlowOptimizer
+version: 1
+with:
+  flow_runner: !MultiFlowRunner
+    with:
+      flows: 
+        - !SingleFlowRunner
+          with:
+            flow_yaml: 'tests/integration/optimizers/flow1.yml'
+            documents: {jsonlines_file}
+            request_size: 1
+            execution_method: 'search_lines'
```

Comment:
if this would be `index_lines`, it would already do what I suggest above.

Reply:
changed, thanks